src/pages/Home.vue

```diff
@@ -38,11 +45,69 @@ export default {
       default: "",
       type: String,
     },
+    program: {
+      /** The program the user falls under. Eg: HaryanaStudents, DelhiStudents */
+      default: "",
+      type: String,
+    },
+    authType: {
+      /** The authentication method used by the user */
+      default: "",
+      type: String,
+    },
+  },
+  data() {
+    return {
+      textObject: {
+        placeholderText: "",
+        displayText: "",
+        addButtonText: "",
+        invalidInputMessage: "",
+        invalidLoginMessage: "",
+        submitButtonText: "",
+      },
+      maxNumberOfInput: 0,
+      inputObject: {
+        basicInputValidationType: "",
+        maxLengthOfId: 0,
+        mode: null,
+        type: null,
+      },
+      dataSourceObject: {
+        name: "",
+        column: "",
+      },
+    };
   },
   computed: {
+    /** Whether multiple ID can be entered */
     isMultipleIDEntryAllowed() {
       return this.redirectTo == "plio";
     },
   },
+  async created() {
+    /** Program name is sent to the API to retrieve all details */
+    var programData = await firebaseAPI.getProgramData(this.program);
+
+    let textObject = programData["text"]["default"];
+    this.textObject["placeholderText"] = textObject["placeholder"];
+    this.textObject["displayText"] = textObject["display"];
+    this.textObject["addButtonText"] = textObject["addButton"];
+    this.textObject["invalidInputMessage"] = textObject["invalid"]["input"];
+    this.textObject["invalidLoginMessage"] = textObject["invalid"]["login"];
+    this.textObject["submitButtonText"] = textObject["submitButton"];
```

Comment:
why not?
```suggestion
    this.textObject = {
        placeholderText: textObject["placeholder"],
        ...
    }
```

Reply:
I don't think there's need for that as well. So what you've done here is you got a programData object from the API call to firebase. After getting it, you're extracting things one by one and putting them into a local variable called textObject, and then passing it to the IdEntry component.

Why not just pass the whole object directly to the IdEntry component without all this parsing? 
If there's some null handling to be done, some edge cases to manage, you can do that here but this is just picking up from one place and putting it into another no?

Same with all the other objects below. All you would need to do is change some variables names in your IdEntry.vue